src/include/migraphx/op/pooling.hpp

```diff
@@ -40,10 +41,18 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace op {
 
+// The Pooling operator mostly follows the specifications for the Onnx pooling op.
+// It assumes an NCHW layout, extended to support any number of spatial dimensions
+// from 1 on up; dimensions are <batch index, channels, spatial dimensions...>
+//
 struct pooling
 {
+    //  Members mode, ceil_mode, padding_mode are all separate concepts with similar names.
     pooling_mode mode = {pooling_mode::average};
 
+    // If the input has rank other than 4 then padding, stride, lengths must all be specified
+    // since the defaults have 2-dimensions.
+
```

Comment:
I don't think it is.  It can be worked out by studying the code, but it's not standard at all in software interfaces that whether certain attributes have to be given at all is dependent on the value of a fourth input.  It's worth pointing out.